odoo/addons/base/models/ir_asset.py

```diff
@@ -0,0 +1,334 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import os
+
+from glob import glob
+from logging import getLogger
+
+from odoo import api, fields, http, models
+
+
+_logger = getLogger(__name__)
+
+SCRIPT_EXTENSIONS = ['js']
+STYLE_EXTENSIONS = ['css', 'scss', 'sass', 'less']
+TEMPLATE_EXTENSIONS = ['xml']
+DEFAULT_SEQUENCE = 16
+
+# Default directive:  `file_path` or `('append', 'file_path')`
+APPEND_DIRECTIVE = 'append'
+# `('include', 'bundle')`
+INCLUDE_DIRECTIVE = 'include'
+# `('prepend', 'file_path')`
+PREPEND_DIRECTIVE = 'prepend'
+# `('remove', 'file_path')`
+REMOVE_DIRECTIVE = 'remove'
+
+# `('after', 'target_path', 'file_path')`
+AFTER_DIRECTIVE = 'after'
+# `('before', 'target_path', 'file_path')`
+BEFORE_DIRECTIVE = 'before'
+# `('replace', 'target_path', 'file_path')`
+REPLACE_DIRECTIVE = 'replace'
+
+DIRECTIVES_WITH_TARGET = [AFTER_DIRECTIVE, BEFORE_DIRECTIVE, REPLACE_DIRECTIVE]
+
+
+def fs2web(path):
+    """Converts a file system path to a web path"""
+    return '/'.join(path.split(os.path.sep))
+
+def get_paths(path_def, extensions, manifest_cache=None):
+    """
+    Returns a list of file paths matching a given glob (path_def) as well as
+    the addon targetted by the path definition. If no file matches that glob,
+    the path definition is returned as is. This is either because the glob is
+    not correctly written or because it points to an URL.
+
+    :param path_def: the definition (glob) of file paths to match
+    :param extensions: a list of extensions that found files must match
+    :returns: a tuple: the addon targetted by the path definition [0] and the
+        list of glob files matching the definition [1] (or the glob itself if
+        none). Note that these paths are filtered on the given `extensions`.
+    """
+    if manifest_cache is None:
+        manifest_cache = http.addons_manifest
```

Comment:
This parameter is not documented.
Is there a reason to not use `http.addons_manifest`?  (Maybe testing?)